opencog/spacetime/Temporal.cc

```diff
@@ -64,8 +64,8 @@ void Temporal::init(unsigned long a, unsigned long b, bool normal) throw (Invali
             throw InvalidParamException(TRACE_INFO,
                                         "Cannot create a Temporal (normal-distribution) with the variance (%lu) greater than the mean (%lu). This causes negative lower bound.", b,  a);
         }
-        unsigned long long sum = (unsigned long long) a + b;
-        if (sum > (unsigned long long) UINT_MAX) {
+        octime_t long sum = (octime_t long) a + b;
```

Comment:
Ick.  even on a 64-bit machine, UINT_MAX is defined to be 2^32-1    Since opencog is unusable on a 32-bit cpu, why not just define octime_t to be a 64-bit int?   Its not like you are saving any space by sing 32-bit ints ...